File: app/src/main/java/org/iggg/anaqram/CharBox.java
package org.iggg.anaqram;

class CharBox {
    private char value;
    private boolean flag;
    private final static char defaultValue = '？';

    CharBox(char c) {
        value = c;
        flag = false;
    }

    void setFlag() {
        flag = true;
    }

    @Override
    public String toString() {
        return String.valueOf(flag ? value : defaultValue);
    }
}

File: app/src/main/java/org/iggg/anaqram/CharBox.kt
package org.iggg.anaqram

internal class CharBox(private val value: Char = defaultValue) {
    private var opened: Boolean = false

    fun open() {
        opened = true
    }

    fun reset() {
        opened = false
    }

    override fun toString(): String =
            (if (opened) value else defaultValue).toString()

    companion object {
        private val defaultValue = '？'
    }
}
